lighthouse-core/audits/byte-efficiency/unminified-css.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ByteEfficiencyAudit = require('./byte-efficiency-audit');
+
+const IGNORE_THRESHOLD_IN_PERCENT = 5;
+const IGNORE_THRESHOLD_IN_BYTES = 512;
+
+const ESCAPE_SLASH = '\\';
+
+/**
+ * @fileOverview
+ */
+class UnminifiedCSS extends ByteEfficiencyAudit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'unminified-css',
+      description: 'Minify CSS',
+      informative: true,
+      helpText: 'Minifying CSS files can reduce network payload sizes.' +
+        '[Learn more](https://developers.google.com/speed/docs/insights/MinifyResources).',
+      requiredArtifacts: ['Styles', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * Computes the total length of the meaningful tokens (CSS excluding comments and whitespace).
+   *
+   * @param {string} content
+   * @return {number}
+   */
+  static computeTokenLength(content) {
+    let totalTokenLength = 0;
+    let isInComment = false;
+    let isInString = false;
+    let stringStarter = null;
+
+    for (let i = 0; i < content.length; i++) {
+      const char = content.charAt(i);
+      const nextChar = content.charAt(i + 1);
+      const chars = char + nextChar;
```

Comment:
might be worth doing perf testing of these.

whats the runtime cost of this audit on a decent size site?

Reply:
test1 https://jsperf.com/charat-master/1 proves winner is 
```
const chars = content.substr(i, 2)
const char = chars.charAt(0)
```

equals is 100x faster than regex for those two lines too 👍 

total runtime on sites with lots of css (i.e. that actually trigger savings) was ~50ms